src/main/java/org/sunbird/operationalreports/service/OperationalReportServiceImpl.java

```diff
@@ -73,82 +76,96 @@ public class OperationalReportServiceImpl implements OperationalReportService {
+                    if (!CollectionUtils.isEmpty(contents)) {
+                        for (Map<String, Object> content : contents) {
+                            rootOrgId = (String) content.get(Constants.ROOT_ORG_ID);
+                            rootOrgIds.add(rootOrgId);
+                        }
+                    }
+                    if (rootOrgIds.size() != 1) {
+                        logger.error("Failed to grant access due to different org.");
+                        throw new HttpClientErrorException(HttpStatus.BAD_REQUEST, "Failed to grant access due to different org.");
+                    }
+                    Map<String, Object> mdoAdminData = userUtilityService.getUsersReadData(mdoLeaderUserId, null, null);
+                    List<String> mdoAdminRoles = (List<String>) mdoAdminData.get(Constants.ROLES);
+                    if (!mdoAdminRoles.contains(Constants.MDO_ADMIN)) {
+                        throw new HttpClientErrorException(HttpStatus.BAD_REQUEST, "The Grantee doesn't have MDO_ADMIN role");
+                    }
+                    if (mdoAdminRoles.contains(Constants.MDO_REPORT_ACCESSOR)) {
+                        Map<String, Object> mdoReportAccessDetails = (Map<String, Object>) getReportAccessDetails(userOrgId).get(mdoAdminUserId);
+                        Date expiryDate = (Date) mdoReportAccessDetails.get(Constants.USER_REPORT_EXPIRY_DATE);
+                        if (expiryDate.compareTo(parseDateFromString(reportExpiryDateRequest)) == 0) {
+                            throw new HttpClientErrorException(HttpStatus.BAD_REQUEST, "The Grantee already has MDO_REPORT_ACCESSOR role till : " + reportExpiryDateRequest);
+                        }
+                        this.updateReportAccessExpiry(mdoAdminUserId, rootOrgId, reportExpiryDateRequest);
+                        response.getResult().put(Constants.STATUS, Constants.SUCCESS);
+                        response.getResult().put(Constants.MESSAGE, "Report access has been granted successfully");
+                        break;
+                    }
+                    mdoAdminRoles.add(Constants.MDO_REPORT_ACCESSOR);
+                    Map<String, Object> assignRoleReq = new HashMap<>();
+                    Map<String, Object> roleRequestBody = new HashMap<>();
+                    roleRequestBody.put(Constants.ORGANIZATION_ID, rootOrgId);
+                    roleRequestBody.put(Constants.USER_ID, mdoAdminUserId);
+                    roleRequestBody.put(Constants.ROLES, mdoAdminRoles);
+                    assignRoleReq.put(Constants.REQUEST, roleRequestBody);
+                    outboundRequestHandlerService.fetchResultUsingPost(serverConfig.getSbUrl() + serverConfig.getSbAssignRolePath(), assignRoleReq,
+                            headersValue);
+                    Map<String, Object> primaryKeyMap = new HashMap<>();
+                    primaryKeyMap.put(Constants.USER_ID_LOWER, mdoAdminUserId);
```

Comment:
No need of these lines -- just call the private method which is used to update the record.